Utilities/File.cpp

```diff
@@ -1898,13 +1903,72 @@ bool fs::file::strict_read_check(u64 _size, u64 type_size) const
 	return true;
 }
 
+std::string fs::get_executable_path()
+{
+#if defined(_WIN32)
+	wchar_t buffer[32767];
+	GetModuleFileNameW(nullptr, buffer, sizeof(buffer) / 2);
+	return wchar_to_utf8(buffer);
+#elif defined(__APPLE__)
+	char bin_path[PATH_MAX];
+	uint32_t bin_path_size = sizeof(bin_path);
+	if (_NSGetExecutablePath(bin_path, &bin_path_size) != 0)
+	{
+		sys_log.error("Failed to find app binary path");
+		return {};
+	}
+
+	return std::filesystem::path(bin_path).parent_path().parent_path().parent_path().string();
+#else
+	if (const char* appimage_path = ::getenv("APPIMAGE"))
+	{
+		sys_log.notice("Found AppImage path: %s", appimage_path);
+		return std::string(appimage_path);
+	}
+
+	sys_log.warning("Failed to find AppImage path");
+
+	char exe_path[PATH_MAX];
+	const ssize_t len = ::readlink("/proc/self/exe", exe_path, sizeof(exe_path) - 1);
+
+	if (len == -1)
+	{
+		sys_log.error("Failed to find executable path");
+		return {};
+	}
+
+	exe_path[len] = '\0';
+	sys_log.trace("Found exec path: %s", exe_path);
```

Comment:
Don't put log here, use std::printf or std::cout or std::cerr or something that doesn't involve the main logging utils.
Because filesystem functions need to be pure and not relying on any dependency to be initialized such as log.